is-file-ignored.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+const path = require('path');
+const fs = require('fs');
+const minimatch = require('minimatch'); // flow-uncovered-line
+
+const getIgnoredPatterns = (fileContents /*: string*/) => {
+    return fileContents
+        .split('\n')
+        .map(line => {
+            if (line.startsWith('#')) {
+                return null;
+            }
+            if (line.startsWith('"')) {
+                throw new Error('Quoted patterns not yet supported, sorry');
+            }
+            if (!line.trim()) {
+                return null;
+            }
+            const [pattern, ...attributes] = line.trim().split(' ');
+            if (attributes.includes('binary') || attributes.includes('linguist-generated=true')) {
```

Comment:
I found the naming of this function to be confusing - it really has nothing to do with "ignored" files (in the `.gitignore` sense) but instead it ignores binary and generated files (which is useful, but different). I think naming and/or some comments may help to clear things up!